crates/oxc_prettier/src/format/block.rs

```diff
@@ -51,22 +51,33 @@ pub(super) fn print_block_body<'a>(
     is_root: bool,
 ) -> Option<Doc<'a>> {
     let has_directives = directives.is_some_and(|directives| !directives.is_empty());
-    let has_body = stmts.iter().any(|stmt| !matches!(stmt, Statement::EmptyStatement(_)));
+    let stmts = Vec::from_iter_in(
+        stmts.iter().filter(|stmt| !matches!(stmt, Statement::EmptyStatement(_))),
+        p.allocator,
+    );
```

Comment:
I'm not sure about reallocating for this. A large file may have thousands of lines which can be fairly large.

Is it possible to pass iterators around?

Reply:
Iterators can't get the length. It might be better to print directives and statements separately.